boards/arm/nrf52840_pca10090/board.c

```diff
@@ -176,20 +176,16 @@ static int pins_configure(struct device *port, const struct pin_config cfg[],
 {
 	int err;
 
-	/* Write to the pins before configuring them as output,
-	 * to make sure we are driving them to the correct level
-	 * right after they are configured.
-	 */
 	for (size_t i = 0; i < pins; i++) {
-		/* The swiches on the board are active low, so we need
-		 * to negate the IS_ENABLED() value from the tables.
+		/* A given pin controlling the switch needs to be driven
+		 * to the low state to activate the routing indicated by
+		 * the corresponding IS_ENABLED() macro in the table,
+		 * so configure the pin as output with the proper initial
+		 * state.
 		 */
-		err = gpio_pin_set(port, cfg[i].pin, !cfg[i].val);
-		if (err) {
-			return cfg[i].pin;
-		}
-
-		err = gpio_pin_configure(port, cfg[i].pin, GPIO_OUTPUT);
+		u32_t flag = (cfg[i].val ? GPIO_OUTPUT_LOW
+					 : GPIO_OUTPUT_HIGH);
+		err = gpio_pin_configure(port, cfg[i].pin, flag);
```

Comment:
So this new API will configure the pin as output and also set its state to low/high?

Reply:
Yes, that's what those flags do.  Unfortunately there aren't `OUTPUT_INIT_ACTIVE` and `OUTPUT_INIT_INACTIVE`.

Since the pins are documented as being active low, and the new API supports operating on logical level, I think it would be more clear to have this be:

```
unsigned int flags = GPIO_ACTIVE_LOW;

if (cfg[i].pin) {
   flags |= GPIO_OUTPUT_LOW;
} else {
   flags |= GPIO_OUTPUT_HIGH;
}
err = gpio_pin_configure(port, cfg[i].pin, flags);
```
then...

Reply:
> Since the pins are documented as being active low

Well, in fact these pins are not documented as active low. The DK documentation says only about the default destination and the optional destination, activated when a given pin is set as output and driven low or high, respectively. These pins are described as having active-low logic only in the [comment](https://github.com/zephyrproject-rtos/zephyr/blob/7c9482e12d7a929c2486aa13d74b81ecbb8c31f7/boards/arm/nrf52840_pca10090/board.c#L65) before definitions of `pins_on_p0` and `pins_on_p1` tables (and I left that comment as it was, treating it as to be considered in the context of these tables only; should I change it?). They were also mentioned as being active low in the comment that I reworded, and I intentionally got rid of this "are active low" statement.

Reply:
OK then.

Reply:
This comment is still in the code, which is part of why I asked for the change:
```
 * The switches have active-low logic, so when writing to the port we will
 * need to invert the value to match the IS_ENABLED() logic.
```

However that's clear enough, so I'm ok with using the negative logic physical value.